### PR TITLE
Decouple pipeline style from strategy

### LINKED ISSUE #699: Separate top level dataflow concept from layer strategy
While updating the configuration, I think we should separate out the concept of top-level dataflow vs pipeline from the layer strategy. Currently, the `_validate_conv_strategy` could inadvertently change a Dense layer in the model from Latency to Resource. I think the concepts are different enough that they should have different configuration parameters.

### PR DESCRIPTION
**Pull request by vloncar**
# Description

We control the top function pipeline style with the strategy, and while this works, sometimes the change to the top pipelining is forced by a layer (the conv layer) and as a consequence this causes all other layers to use `resource` strategy which is unintended. Decoupling the strategy from the pipeline style fixes this. The explicitly set resource strategy still forces the `dataflow` style to be used. The configuration can be explicitly set with `PipelineStyle` (values can be `pipeline` and `dataflow`), but normally the users would still continue using the `Strategy` config parameter. Fixes #699 and #759.

## Type of change

- [x] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Tests

I didn't add any extra tests. The best way to test this is to verify the generated HLS has the correct pragmas and `parameters.h`. I manually tested a whole lot of combinations of this. Checking the IR in python may also be possible, but I'm not sure if this will capture all combinations.

## Checklist

- [x] I have read the [guidelines for contributing](https://github.com/fastmachinelearning/hls4ml/blob/main/CONTRIBUTING.md).
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have installed and run `pre-commit` on the files I edited or added.
- [ ] I have added tests that prove my fix is effective or that my feature works.

### CODE REVIEW

File: hls4ml/model/graph.py [hunk 1]
```diff
@@ -274,45 +277,48 @@ def _parse_hls_config(self):
                     self.layer_name_compression[layer_name.lower()] = bool(compression)
 
     def _validate_hls_config(self):
-        use_resource = False
-        if self.model_strategy.lower() == 'latency' and self.model_compression:
+        use_dataflow = False
+        if self.pipeline_style.lower() == 'pipeline' and self.model_compression:
             print('WARNING: Compression enabled while model strategy set to "Latency".')
```
Comment (jmitrevs) [inline hls4ml/model/graph.py:282]: The warning needs updating

**Comment by jmitrevs** [comment]
Do we understand why the tests failed? Otherwise, the requested change is quite minor.

**Comment by vloncar** [comment]
Not sure why the tests failed, but certainly it is not due to the changes in the code. Since this started happening recently and we didn't change the CI configuration, it may be due to the increasing number of parallel tests each creating a new environment. Not sure why it doesn't deterministically fail. We need to investigate further. You can rerun the tests or run them locally to be sure that indeed nothing got broken. I ran locally, it all passed.

**Review by jmitrevs — APPROVED** [review]

**Comment by jmitrevs** [comment]
Unless people object, I will merge this tomorrow. But is this for 0.8 or 0.7.1?

**Comment by vloncar** [comment]
How about we merge #780, fix the capitalization issue observed in #782, release that as 0.7.1 and then start merging for 0.8.0 with this PR?

**Pull request merged**